Overhaul documentation

gfontenot opened the pull request:
- Improve help doc significantly (I hope). Use more sections and write
  more about each command/option.
  - Point people to the help doc from within the README instead of
    duplicating info. The README should be an overview of features, and
    the help document should be where people can find out more.

ping @gabebw for his ever-helpful prose advice.

gfontenot committed:
Overhaul documentation

 - Improve help doc significantly (I hope). Use more sections and write
   more about each command/option.
 - Point people to the help doc from within the README instead of
   duplicating info. The README should be an overview of features, and
   the help document should be where people can find out more.

gabebw commented:
While you're in there, can you change this?

> - `XBuild` will build the project
> - `XTest` will test the project in the iOS simulator
 
I'd like `:XBuild` and `:XTest` so people know they're commands.

gabebw commented on README.md:
```diff
@@ -28,23 +28,10 @@ have it installed, `xcodebuild.vim` will pipe all `xcodebuild` output through
 
-If you'd like to customize the output from `xcpretty`, you can do so by
-setting a couple of global variables in your `vimrc`:
+For customization options, see the [included help doc][help] (`:h xcodebuild`
```
I like spelling everything out in docs: `:help`

gfontenot committed:
Remove reference to the iOS simulator

This is outdated info (because I'm moving so _fast_). We shouldn't
confuse people by specifying that tests are run in the iOS simulator,
and should instead just say that tests are run.

gabebw commented on README.md:
```diff
@@ -65,7 +52,12 @@ This is useful for using `xcodebuild.vim` with other plugins such as
 
-xcodeproj.vim is copyright © 2015 Gordon Fontenot It is free software, and may be
-redistributed under the terms specified in the `LICENSE` file.
+xcodeproj.vim is copyright © 2015 Gordon Fontenot It is free software, and may
```
Period after your last name

gabebw commented on doc/xcodebuild.txt:
```diff
@@ -1,31 +1,139 @@
-CONTENTS                               *xcodebuild-contents*
+CONTENTS                                                         *XCB-Contents*
 
+      1. About.......................... |XCB-About|
```
I think every help section should start with `xcodebuild`, like [fugitive](https://github.com/tpope/vim-fugitive/blob/master/doc/fugitive.txt) does.

That way `:help xcodebuild<TAB>` shows everything, instead of only the parts that start with xcodebuild.

Maybe keep `XBuild` and `XTest` the same, though I think `:help XBuild<TAB>` will find it even if it's `xcodebuild-XBuild`.

gfontenot replied:
should I rename the global sections too? `xcodebuild-xcodebuild_xcpretty_testing_flags` is pretty verbose.

christoomey replied:
I agree with the general idea of namespacing them to make help lookup easier. In this case I think having the single leading `xcodebuild` is sufficient, rather than repeated.

gabebw commented on doc/xcodebuild.txt:
```diff
@@ -1,31 +1,139 @@
 
 ==============================================================================
-INTRODUCTION                           *xcodebuild*
+ABOUT (1)                                                           *XCB-About*
```
What's the number for?

gfontenot replied:
No clue. Stole this from the venerable @christoomey

christoomey replied:
Mine certainly was stolen from someone else's. I think the numbers are just numbers, whereas the `~` a bit lower down does have some other effect either in just syntax highlighting, or semantic (tag?) value.

gabebw commented on doc/xcodebuild.txt:
```diff
@@ -1,31 +1,139 @@
+the output from `xcodebuild`.
+
+This plugin was written by Gordon Fontenot[2]. Bugs and feature requests are
+welcomed, and can be posted on the GitHub repo[3].
```
Thank you for spelling it GitHub.

gabebw commented on doc/xcodebuild.txt:
```diff
@@ -1,31 +1,139 @@
+
+------------------------------------------------------------------------------
+                                                                       *XBuild*
+2.1 XBuild~
```
What does the tilde do?

gfontenot replied:
It...highlights it different? I'm sure there's a semantic reason for this. Again, stole this from the legendary @christoomey, so I'm not sure.

christoomey replied:
Looks like it makes them an `hl<helpHeader>`. For me that makes them blueish, which I kinda like...

gabebw commented on doc/xcodebuild.txt:
```diff
@@ -1,31 +1,139 @@
+
+Run the project tests through `xcodebuild`.
+
+The first time this command is run, there will be a small delay as we parse
```
we parse -> the plugin parses, I think

gabebw commented on doc/xcodebuild.txt:
```diff
@@ -1,31 +1,139 @@
+Run the project tests through `xcodebuild`.
+
+The first time this command is run, there will be a small delay as we parse
+the sdk information for the scheme, and might be a delay while we parse the
```
sdk -> SDK

and nix the "we"  here too

Unless you come over to every user's house and type for them? If so, I'd add another section to this help document listing legal disclaimers.

gabebw commented on doc/xcodebuild.txt:
```diff
@@ -1,31 +1,139 @@
+                                                                     *xcpretty*
+2.3 xcpretty~
+
+If `xcpretty`[1] is installed and is available in your `PATH`, `xcodebuild.vim`
```
`$PATH`?

gfontenot committed:
Clarify that `XBuild` and `XTest` are commands

gfontenot committed:
Use full :help command

gfontenot committed:
Add punctuation in License section

gfontenot committed:
Prefix help sections consistently with xcodebuild

This will help with autocomplete and discoverability.

gfontenot committed:
Documentation updates

Nerdy grammar shit.

gfontenot commented:
Thanks, fellas. This is much nicer now.

gabebw commented on doc/xcodebuild.txt:
```diff
@@ -1,31 +1,140 @@
-CONTENTS                               *xcodebuild-contents*
+CONTENTS                                                 *xcodebuild--Contents*
 
+      1. About.......................... |xcodebiuld-About|
```
Typo: `xcodebiuld`

gfontenot replied:
ugh, thanks. I fixed these typos on master.

gabebw commented on doc/xcodebuild.txt:
```diff
@@ -1,31 +1,140 @@
+      2. Usage ......................... |xcodebuild-Usage|
+        2.1  ............................. |xcodebuild-:XBuild|
+        2.2  ............................. |xcodebuild-:XTest|
+        2.3  ............................. |scodebiuld-xcpretty|
```
Typo: `scodebiuld`
